src/main/java/org/apache/parquet/hadoop/BdecParquetWriter.java

```diff
@@ -35,6 +36,7 @@
 public class BdecParquetWriter implements AutoCloseable {
   private final InternalParquetRecordWriter<List<Object>> writer;
   private final CodecFactory codecFactory;
+  private final AtomicLong rowsWritten = new AtomicLong(0);
```

Comment:
Can it be just a `volatile`? This goes back to my previous question about races. If there is no concurrent access (but non-overlapping access from different threads), then `volatile` would be sufficient.

Reply:
In fact, in this case, we don't even need `volatile`. The field is accessed either within a lock, or in a BG thread with happens-before relationship with the spawning thread.